compiler/ast.nim

```diff
@@ -477,6 +477,7 @@ type
     nfDefaultRefsParam # a default param value references another parameter
                        # the flag is applied to proc default values and to calls
     nfExecuteOnReload  # A top-level statement that will be executed during reloads
+    nfOverloadResolve # return resolved `foo` in `foo(args)`
```

Comment:
I really dislike these "node flags" and adding more makes their problems worse. Is there no other way to accomplish the same?

Reply:
> resolveSymbol(foo($a)) where, if you apply flags.incl efOverloadResolve on the whole foo($a) expression, the argument $a to foo would be subject to efOverloadResolve (skipping eval) instead of being evaluated.

Well you don't have to propagate efOverloadResolve recursively. And as a node flag the problem is the very same: Either you propagate this flag or you don't. IMO it would be a far better solution.

Reply:
done, I'm now using `efOverloadResolve` instead